From 8a29c1363fd5a5418f1d03c7efe175057cbe558b Mon Sep 17 00:00:00 2001
From: Wei-Chiu Chuang <weichiu@apache.org>
Date: Mon, 16 Oct 2017 12:57:48 -0700
Subject: [PATCH 2706/2748] HDFS-12485. expunge may fail to remove trash from
 encryption zone. Contributed by Wei-Chiu Chuang.

(cherry picked from commit 8dbc8909c92d502d10a7f94d1de3171878a43b04)
(cherry picked from commit a596c6f5eb553cf38123d61b844696ddbec99be0)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DistributedFileSystem.java
(cherry picked from commit 843b8b7b969b96d94731efa6e12c54461f3d02ce)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DistributedFileSystem.java
	hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestTrashWithSecureEncryptionZones.java

Change-Id: I68e12ae6fc8448c126208eef7557c559d85cd651
(cherry picked from commit 83163cf83fa439cf6f9e1240d182b90b558c82b2)
---
 .../apache/hadoop/hdfs/DistributedFileSystem.java  |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DistributedFileSystem.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DistributedFileSystem.java
index a1e8981..608ef78 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DistributedFileSystem.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DistributedFileSystem.java
@@ -2421,8 +2421,7 @@ public Path getTrashRoot(Path path) {
             }
           }
         } else {
-          Path userTrash = new Path(ezTrashRoot, System.getProperty(
-              "user.name"));
+          Path userTrash = new Path(ezTrashRoot, dfs.ugi.getShortUserName());
           if (exists(userTrash)) {
             ret.add(getFileStatus(userTrash));
           }
-- 
1.7.9.5

